bazel/expanded_template/BUILD

```diff
@@ -0,0 +1,5 @@
+cc_binary(
+    name = "expand_template",
+    srcs = ["expand_template.cc"],
+    visibility = ["//visibility:public"],
```

Comment:
Would it make sense to make this target private?

Reply:
That doesn't work, because it's an implicit dependency of the gflags rule. The dependency is resolved wrt to the BUILD file that uses gflags.

I wish there were a way to express that in Blaze, but there isn't :(

Reply:
Ok. Then just to help me understand potential repercussions of this.

Would `load("//bazel/expanded_template:expanded_template.bzl", "expanded_template")` load this extension if used in a dependent `BUILD` file? Is there a potential of name conflict with another library?

Reply:
The answer might be [here](https://docs.bazel.build/versions/master/build-ref.html#labels) in the Bazel docs. The fully qualified target name would include `@gflags` prefix as I understand.

Reply:
The repercussion is that someone can peek into our private `.bzl` rule and `cc_binary` and use them, and then we might break them when we update them. I mean, more fool them for depending on what is clearly internal stuff, but they could.

There is definitely no name conflict issue. The _visibility_ is resolved wrt the using package, but not the name. It's not like the name is relative to the including package or anything like that.

(FYI I used to work on the Blaze team)